src/discordtipbot.cr

```diff
@@ -1,44 +1,37 @@
-require "raven"
-require "logger"
-require "pg"
-require "pg/pg_ext/big_decimal"
-require "discordcr"
-require "big"
-require "big/json"
-
-require "./discordtipbot/*"
-
-unless ENV["TIPBOT_ENV"]? == "test"
-  puts "No Config File specified! Exiting!" if ARGV.size == 0
-  exit if ARGV.size == 0
-
-  Raven.configure do |config|
-    config.async = true
-  end
-
-  Raven.capture do
-    log = Logger.new(STDOUT)
+class DiscordTipBot
+  def initialize
+    abort "No Config File specified! Exiting!" if ARGV.size == 0
 
-    # Set your logger level here
-    log.level = Logger::DEBUG
+    @log = Logger.new(STDOUT)
 
-    log.debug("Tipbot network getting started")
+    load_config
 
-    log.debug("Attempting to read config from \"#{ARGV[0]}\"")
-    config = File.open(ARGV[0], "r") do |file|
-      Array(Config).from_json(file)
+    Raven.configure do |raven_config|
+      raven_config.async = true
     end
-    log.info("read config from \"#{ARGV[0]}\"")
 
-    log.debug("starting forking")
-    config.each do |x|
-      raven_spawn(name: "#{x.coinname_full} Bot") do
-        Controller.new(x, log)
+    Raven.capture do
+      # Set your log level here
+      @log.level = Logger::DEBUG
+
+      @log.debug("Tipbot network getting started")
+
+      @log.debug("starting forking")
+      Config.current.each do |name, config|
+        raven_spawn(name: "#{name} Bot") do
+          Controller.new(config, @log)
+        end
       end
+      @log.debug("finished forking")
+
+      @log.info("All bots should be running now")
     end
-    log.debug("finished forking")
+    sleep
```

Comment:
I think its really surprising / awkward that `Foo.new` is a blocking operation.

You could:

1. Change this whole method accordingly to `DiscordTipBot.run` instead, since you will never actually instance the object you just allocated
2. `sleep` elsewhere (`launcher.cr`)